api_functional_v4_test.go

```diff
@@ -497,6 +498,86 @@ func TestGetObjectClosedTwice(t *testing.T) {
 	}
 }
 
+// Test removing multiple objects with MultiRemove API
+func TestRemoveMultipleObjects(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping function tests for short runs")
+	}
+
+	// Seed random based on current time.
+	rand.Seed(time.Now().Unix())
+
+	// Instantiate new minio client object.
+	c, err := New(
+		"s3.amazonaws.com",
+		os.Getenv("ACCESS_KEY"),
+		os.Getenv("SECRET_KEY"),
+		true,
+	)
+
+	if err != nil {
+		t.Fatal("Error:", err)
+	}
+
+	// Set user agent.
+	c.SetAppInfo("Minio-go-FunctionalTest", "0.1.0")
+
+	// Enable tracing, write to stdout.
+	// c.TraceOn(os.Stderr)
+
+	// Generate a new random bucket name.
+	bucketName := randString(60, rand.NewSource(time.Now().UnixNano()), "minio-go-test")
+
+	// Make a new bucket.
+	err = c.MakeBucket(bucketName, "us-east-1")
+	if err != nil {
+		t.Fatal("Error:", err, bucketName)
+	}
+
+	r := bytes.NewReader(bytes.Repeat([]byte("a"), 8))
+
+	// Multi remove of 1100 objects
+	nrObjects := 1100
+
+	// Prepare objects and upload them
+	objects := make([]string, nrObjects)
```

Comment:
Perhaps rather than buffering like this you can show that as soon as you have objectName you are writing to channel. 

```
go func() 
      defer close(objectsCh) 
       for i := 0; i < nrObjects; i++ {
 		objectName := "sample" + strconv.Itoa(i) + ".txt"
 		_, err = c.PutObject(bucketName, objectName, r, "application/octet-stream")
 		if err != nil {
 			t.Fatal("Error: PutObject shouldn't fail.")
 		}
 		objectCh <- objectName
 	}
}()
```

what do you think?